Convert CRLF for script bin files

lilyinstarlight opened the pull request:
After we started generating our own bin links in #302, we did not provide the line-ending normalization that npm did

This PR corrects that to have similar behavior to before for packages with CRLF line-endings

marsam commented on nix/node-env.nix:
```diff
@@ -535,6 +535,8 @@ let
                 file="$(readlink -f "$i")"
                 chmod u+rwx "$file"
+                isScript "$file" || continue
+                dos2unix "$file"
```
I think you can use sed:
```suggestion
                sed -i 's/\r*$//' "$file"
```
Also, just curious, doesn't `patchShebangsAuto` automatically patches these files?

lilyinstarlight replied:
I'll test that sed command when I'm back at my computer. That would be a lot lighter for sure

And I don't think `patchShebangs` fixes line-endings but I'd have to go check the code again

lilyinstarlight replied:
Okay, I've converted it to sed, making this a less invasive change. Thank you for the recommendation!

marsam replied:
>And I don't think `patchShebangs` fixes line-endings but I'd have to go check the code again

Sorry, I mean `patchShebangsAuto` automatically patches the interpreters in `$out/bin`, and is added [by default to the fixup phase](https://nixos.org/manual/nixpkgs/stable/#patch-shebangs.sh). So we don't need to call patchShebangs ourselves, and this bit could be something like:
```bash
# Create symlink to the deployed executable folder, if applicable
if [ -d "$out/lib/node_modules/.bin" ]
then
    ln -s $out/lib/node_modules/.bin $out/bin
    find $out/bin -type f -print0 | xargs -0 sed -i 's/\r$//'
fi
```

lilyinstarlight replied:
Ah, I misunderstood. I'll try to see if we can get by without it then (though that call predated this PR)

Note that we still want to do `isScript` detection because that sed command can do bad things to binaries :)

lilyinstarlight replied:
Yes, it turns out to be unnecessary. I've included removing `patchShebangs` there as a drive-by change, but I would like to avoid making too many unrelated changes in this PR, if that's okay

Does what I pushed look good to you now @marsam?

marsam replied:
Yes, LGTM. Thanks!

lilyinstarlight committed:
Convert CRLF for script bin files

After we started generating our own bin links, we did not provide the
line-ending normalization that npm did. This corrects that to have
similar behavior to before for packages with CRLF line-endings.

This also has a drive-by change to remove the unnecessary
`patchShebangs` due to the `patchShebangsAuto` hook running and finding
these files just fine now.
